In [21]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

Neste caderno notebook, o objetivo é realizar uma análise de dados sobre o preço da gasolina.

*Os datasets estão separados por quetão didática, mas segue abaixo o link:https://www.kaggle.com/datasets/matheusfreitag/gas-prices-in-brazil

In [22]:
df1 = pd.read_csv("Datasets/gasolina_2000+.csv", index_col=0)
df2 = pd.read_csv("Datasets/gasolina_2010+.csv", index_col=0, low_memory=False)

df = pd.concat([df1, df2])

In [23]:
df.head()

DATA INICIAL  DATA FINAL        REGIÃO              ESTADO  \
0   2004-05-09  2004-05-15  CENTRO OESTE    DISTRITO FEDERAL   
1   2004-05-09  2004-05-15  CENTRO OESTE               GOIAS   
2   2004-05-09  2004-05-15  CENTRO OESTE         MATO GROSSO   
3   2004-05-09  2004-05-15  CENTRO OESTE  MATO GROSSO DO SUL   
4   2004-05-09  2004-05-15      NORDESTE             ALAGOAS   

            PRODUTO  NÚMERO DE POSTOS PESQUISADOS UNIDADE DE MEDIDA  \
0  ETANOL HIDRATADO                           127              R$/l   
1  ETANOL HIDRATADO                           387              R$/l   
2  ETANOL HIDRATADO                           192              R$/l   
3  ETANOL HIDRATADO                           162              R$/l   
4  ETANOL HIDRATADO                           103              R$/l   

   PREÇO MÉDIO REVENDA  DESVIO PADRÃO REVENDA  PREÇO MÍNIMO REVENDA  \
0                1.288                  0.016                  1.19   
1                1.162                  0.114                  0.89   
2                1.389                  0.097                  1.18   
3                1.262                  0.070                  1.09   
4                1.181                  0.078                  1.05   

   PREÇO MÁXIMO REVENDA MARGEM MÉDIA REVENDA  COEF DE VARIAÇÃO REVENDA  \
0                 1.350                0.463                     0.012   
1                 1.449                0.399                     0.098   
2                 1.760                0.419                     0.070   
3                 1.509                0.432                     0.055   
4                 1.400                 0.24                     0.066   

  PREÇO MÉDIO DISTRIBUIÇÃO DESVIO PADRÃO DISTRIBUIÇÃO  \
0                    0.825                       0.11   
1                    0.763                      0.088   
2                     0.97                      0.095   
3                     0.83                      0.119   
4                    0.941                      0.077   

  PREÇO MÍNIMO DISTRIBUIÇÃO PREÇO MÁXIMO DISTRIBUIÇÃO  \
0                    0.4201                    0.9666   
1                    0.5013                      1.05   
2                    0.5614                     1.161   
3                    0.5991                   1.22242   
4                    0.7441                    1.0317   

  COEF DE VARIAÇÃO DISTRIBUIÇÃO  
0                         0.133  
1                         0.115  
2                         0.098  
3                         0.143  
4                         0.082

__Conhecendo o conjunto de dados__

In [24]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 120823 entries, 0 to 120822
Data columns (total 18 columns):
 #   Column                         Non-Null Count   Dtype  
---  ------                         --------------   -----  
 0   DATA INICIAL                   120823 non-null  object 
 1   DATA FINAL                     120823 non-null  object 
 2   REGIÃO                         120823 non-null  object 
 3   ESTADO                         120823 non-null  object 
 4   PRODUTO                        120823 non-null  object 
 5   NÚMERO DE POSTOS PESQUISADOS   120823 non-null  int64  
 6   UNIDADE DE MEDIDA              120823 non-null  object 
 7   PREÇO MÉDIO REVENDA            120823 non-null  float64
 8   DESVIO PADRÃO REVENDA          120823 non-null  float64
 9   PREÇO MÍNIMO REVENDA           120823 non-null  float64
 10  PREÇO MÁXIMO REVENDA           120823 non-null  float64
 11  MARGEM MÉDIA REVENDA           120823 non-null  object 
 12  COEF DE VARIAÇÃO REVENDA      

_Alteração dos tipos nas colunas

* 0   DATA INICIAL                   120823 non-null  object  Datetime
* 1   DATA FINAL                     120823 non-null  object  Datetime
* 2   REGIÃO                         120823 non-null  object  Correto
* 3   ESTADO                         120823 non-null  object  Correto
* 4   PRODUTO                        120823 non-null  object  Correto
* 5   NÚMERO DE POSTOS PESQUISADOS   120823 non-null  int64   Correto
* 6   UNIDADE DE MEDIDA              120823 non-null  object  Inteiro
* 7   PREÇO MÉDIO REVENDA            120823 non-null  float64 Correto
* 8   DESVIO PADRÃO REVENDA          120823 non-null  float64 Correto
* 9   PREÇO MÍNIMO REVENDA           120823 non-null  float64 Correto
* 10  PREÇO MÁXIMO REVENDA           120823 non-null  float64 Correto
* 11  MARGEM MÉDIA REVENDA           120823 non-null  object  Float
* 12  COEF DE VARIAÇÃO REVENDA       120823 non-null  float64 Correto
* 13  PREÇO MÉDIO DISTRIBUIÇÃO       120823 non-null  object  Float
* 14  DESVIO PADRÃO DISTRIBUIÇÃO     120823 non-null  object  Float
* 15  PREÇO MÍNIMO DISTRIBUIÇÃO      120823 non-null  object  Float
* 16  PREÇO MÁXIMO DISTRIBUIÇÃO      120823 non-null  object  Float
* 17  COEF DE VARIAÇÃO DISTRIBUIÇÃO  120823 non-null  object  Float
* dtypes: float64(5), int64(1), object(12)

In [33]:
df = pd.to_numeric(df[['MARGEM MÉDIA REVENDA', 'PREÇO MÉDIO DISTRIBUIÇÃO', 'DESVIO PADRÃO DISTRIBUIÇÃO', 'PREÇO MÍNIMO DISTRIBUIÇÃO', 'PREÇO MÁXIMO DISTRIBUIÇÃO', 'COEF DE VARIAÇÃO DISTRIBUIÇÃO']])


TypeError: arg must be a list, tuple, 1-d array, or Series

In [27]:
df

DATA INICIAL  DATA FINAL        REGIÃO              ESTADO  \
0        2004-05-09  2004-05-15  CENTRO OESTE    DISTRITO FEDERAL   
1        2004-05-09  2004-05-15  CENTRO OESTE               GOIAS   
2        2004-05-09  2004-05-15  CENTRO OESTE         MATO GROSSO   
3        2004-05-09  2004-05-15  CENTRO OESTE  MATO GROSSO DO SUL   
4        2004-05-09  2004-05-15      NORDESTE             ALAGOAS   
...             ...         ...           ...                 ...   
120818   2021-04-25  2021-05-01         NORTE             RORAIMA   
120819   2021-04-25  2021-05-01           SUL      SANTA CATARINA   
120820   2021-04-25  2021-05-01       SUDESTE           SAO PAULO   
120821   2021-04-25  2021-05-01      NORDESTE             SERGIPE   
120822   2021-04-25  2021-05-01         NORTE           TOCANTINS   

                 PRODUTO  NÚMERO DE POSTOS PESQUISADOS UNIDADE DE MEDIDA  \
0       ETANOL HIDRATADO                           127              R$/l   
1       ETANOL HIDRATADO                           387              R$/l   
2       ETANOL HIDRATADO                           192              R$/l   
3       ETANOL HIDRATADO                           162              R$/l   
4       ETANOL HIDRATADO                           103              R$/l   
...                  ...                           ...               ...   
120818   OLEO DIESEL S10                            12              R$/l   
120819   OLEO DIESEL S10                           100              R$/l   
120820   OLEO DIESEL S10                           886              R$/l   
120821   OLEO DIESEL S10                            33              R$/l   
120822   OLEO DIESEL S10                            18              R$/l   

        PREÇO MÉDIO REVENDA  DESVIO PADRÃO REVENDA  PREÇO MÍNIMO REVENDA  \
0                     1.288                  0.016                 1.190   
1                     1.162                  0.114                 0.890   
2                     1.389                  0.097                 1.180   
3                     1.262                  0.070                 1.090   
4                     1.181                  0.078                 1.050   
...                     ...                    ...                   ...   
120818                4.582                  0.056                 4.510   
120819                4.141                  0.187                 3.838   
120820                4.246                  0.216                 3.769   
120821                4.464                  0.205                 4.190   
120822                4.352                  0.156                 4.199   

        PREÇO MÁXIMO REVENDA MARGEM MÉDIA REVENDA  COEF DE VARIAÇÃO REVENDA  \
0                      1.350                0.463                     0.012   
1                      1.449                0.399                     0.098   
2                      1.760                0.419                     0.070   
3                      1.509                0.432                     0.055   
4                      1.400                 0.24                     0.066   
...                      ...                  ...                       ...   
120818                 4.690             -99999.0                     0.012   
120819                 4.739             -99999.0                     0.045   
120820                 5.199             -99999.0                     0.051   
120821                 4.699             -99999.0                     0.046   
120822                 4.699             -99999.0                     0.036   

       PREÇO MÉDIO DISTRIBUIÇÃO DESVIO PADRÃO DISTRIBUIÇÃO  \
0                         0.825                       0.11   
1                         0.763                      0.088   
2                          0.97                      0.095   
3                          0.83                      0.119   
4                         0.941                      0.077   
...                         ...